server/src/main/scala/drt/server/feeds/edi/EdiFeed.scala

```diff
@@ -0,0 +1,131 @@
+  }
+
+}
+
+
+class EdiFeed(ediClient: EdiClient) extends EdiFeedJsonSupport {
+
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def ediFeedPollingSource(interval: FiniteDuration, startData: String, endDate: String, feedSource: FeedSource)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Source[ArrivalsFeedResponse, Cancellable] = {
+    Source.tick(1 seconds, interval, NotUsed)
+      .mapAsync(1) { _ =>
+        log.info(s"$feedSource Edi feed api call at ${DateTime.now()} from $startData and $endDate ")
+        ediClient.makeRequest(startData, endDate).flatMap { response =>
+          response.status match {
+            case OK => unMarshalResponseToEdiFlightDetails(response).map { flights =>
+              log.info(s"Edi feed status ${response.status} with api call for ${flights.size} flights")
+              ArrivalsFeedSuccess(Flights(ediFlightDetailsToArrival(flights, feedSource)))
+            }
+            case _ => log.warn(s"Edi feed status ${response.status} while api call with response ${response.entity}")
+              Future.successful(ArrivalsFeedFailure(s"Response with status ${response.status} from edi ${response.entity}"))
+          }
+        }.recover { case t =>
+          log.error(s"Edi feed error while api call ${t.getMessage}")
+          ArrivalsFeedFailure(t.getMessage)
+        }
+      }
+  }
+
+
+  def getStatusDescription(status: String): String = status match {
+    case "A" => "Arrival is on block at a stand"
+    case "D" => "Departure off block from a stand"
+    case "F" => "Finals"
+    case "L" => "Landed"
+    case "N" => "Info Due"
+    case "G" => "(Go Around) Overshot the landing"
+    case "P" => "Airborne from preceding airport"
+    case "S" => "Flight is on schedule"
+    case "T" => "Departure is airborne"
+    case "V" => "Arrival diverted away from airport"
+    case "X" => "Cancelled"
+    case "Z" => "Zoned"
+    case "E" => "Estimated"
+    case _ => status
+  }
+
+  def removeFlightNumberChar(flightNumber: String) = {
+    if (flightNumber.matches("[0-9]+[a-zA-Z]"))
+      flightNumber.substring(0, flightNumber.length - 1)
+    else flightNumber
+
+  }
+
+  def ediFlightDetailsToArrival(ediFlightDetailsList: Seq[EdiFlightDetails], feedSource: FeedSource): List[Arrival] = ediFlightDetailsList
+    .filter(_.ArrDeptureCode.toLowerCase == "a")
+    .filter(_.DomsINtlCode.toLowerCase == "i")
+    .map(f = flight => Try {
+      val est = flight.EstimatedDateTime_Zulu.map(SDate(_).millisSinceEpoch).getOrElse(0L)
+      val act = flight.ActualDateTime_Zulu.map(SDate(_).millisSinceEpoch).getOrElse(0L)
+      val actChox = flight.ChocksDateTime_Zulu.map(SDate(_).millisSinceEpoch).getOrElse(0L)
+      Arrival(
+        Operator = Option(Operator(flight.TicketedOperator)),
+        CarrierCode = CarrierCode(flight.AirlineCode_IATA),
+        VoyageNumber = VoyageNumber(removeFlightNumberChar(flight.FlightNumber).toInt),
+        FlightCodeSuffix = None,
```

Comment:
We should use the letter at the end of a flight number for this value

Reply:
make sense